public/scripts/editor/js/fc/project-rename.js

```diff
@@ -0,0 +1,89 @@
+define(function(require) {
+  var $ = require("jquery");
+  var InputField = require("fc/bramble-input-field");
+
+  function toggleComponents(context, isSave) {
+    context.saveButton[isSave ? "hide" : "show"]();
+    context.renameButton[isSave ? "show" : "hide"]();
+    context.titleBar[isSave ? "save" : "edit"]();
+    context.titleBar[isSave ? "removeClass" : "addClass"]("text-input");
+    context.container[isSave ? "addClass" : "removeClass"]("shadow");
+  }
+
+  function persist(title, callback) {
+    var appUrl = this.appUrl;
+    var csrfToken = this.csrfToken;
+
+    if(!this.authenticated) {
```

Comment:
If we're letting them change the title when they aren't authenticated, does that mean that we store that somewhere in case they log in?